test/unit/logger-mixin.test.ts

```diff
@@ -0,0 +1,114 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback-next-extension-starter
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {LoggerMixin} from '../..';
+import {Application, Component} from '@loopback/core';
+import {Constructor} from '@loopback/context';
+
+describe('LoggerMixin', () => {
+  it('mixed class has .logger()', () => {
+    const myApp = new AppWithLogger();
+    expect(typeof myApp.logger).to.be.eql('function');
+  });
+
+  it('binds Logger from constructor', () => {
+    const myApp = new AppWithLogger({
+      loggers: [MyLogger],
+    });
+
+    expectLoggerToBeBound(myApp);
+  });
+
+  it('binds Logger from app.logger()', () => {
+    const myApp = new AppWithLogger();
+
+    expectLoggerToNotBeBound(myApp);
+    myApp.logger(MyLogger);
+    expectLoggerToBeBound(myApp);
+  });
+
+  it('binds user defined component with Logger in constructor', () => {
+    const myApp = new AppWithLogger({
+      loggers: [MyLogger],
+    });
+
+    expectLoggerToBeBound(myApp);
+  });
+
+  it('binds user defined component with logger from .component()', () => {
+    const myApp = new AppWithLogger();
+
+    const boundComponentsBefore = myApp.find('components.*').map(b => b.key);
+    expect(boundComponentsBefore).to.be.empty();
+
+    myApp.component(TestComponent);
+
+    expectComponentToBeBound(myApp, TestComponent);
+    expectLoggerToBeBound(myApp);
+  });
+
+  it('returns instantiated instance of Logger for use', () => {
+    const myApp = new AppWithLogger({
+      loggers: [MyLogger],
+    });
+
+    const logger = myApp.getSync('loggers.MyLogger');
+    expect(logger).to.be.instanceOf(MyLogger);
+  });
+
+  class AppWithLogger extends LoggerMixin(Application) {}
+
+  class MyLogger {
+    // tslint:disable-next-line:no-any
+    log(...args: any[]) {
```

Comment:
In order to avoid those tslint exceptions, I am proposing to define a type alias:

```ts
// tslint:disable-next-line:no-any
type LogArgs = any[];

// usage
log(...args: LogArgs) {
  // ...
}
```